### ISSUE TITLE
Error message of motif_discovery

### ISSUE THREAD

**Issue by YuyaKiguchi** [MicrobialDarkMatter/nanomotif#54]
Thanks for developing a useful tool.
I'm using motif_discovery to find the methylation motifs from metagenomic Bins.

But, I got this error using the following command.
`nanomotif motif_discovery input.bed contig.bin.tsv -t 16 input.nanomotif`

But, I can find the output files of these.
Could you solve the this error? or I don't mind this error?
motifs-raw.tsv
motifs-score-sub.tsv
motifs-score.tsv

Error message
`thread 'polars-0' panicked at /home/conda/feedstock_root/build_artifacts/polars_1717264040812/work/crates/polars-core/src/chunked_array/ops/chunkops.rs:109:17:
implementation error
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Traceback (most recent call last):
  File "/home/kiguchi/miniconda3/envs/nanomotif/bin/nanomotif", line 10, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/kiguchi/miniconda3/envs/nanomotif/lib/python3.12/site-packages/nanomotif/main.py", line 514, in main
    motif_discovery(args)
  File "/home/kiguchi/miniconda3/envs/nanomotif/lib/python3.12/site-packages/nanomotif/main.py", line 303, in motif_discovery
    motifs = find_motifs(args, pileup=pileup, assembly=assembly)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kiguchi/miniconda3/envs/nanomotif/lib/python3.12/site-packages/nanomotif/main.py", line 157, in find_motifs
    motifs = nm.postprocess.remove_noisy_motifs(motifs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kiguchi/miniconda3/envs/nanomotif/lib/python3.12/site-packages/nanomotif/postprocess.py", line 12, in remove_noisy_motifs
    for contig, df in motif_df.groupby("contig"):
  File "/home/kiguchi/miniconda3/envs/nanomotif/lib/python3.12/site-packages/polars/dataframe/group_by.py", line 136, in __next__
    group_data = self.df[self._group_indices[self._current_index], :]
                 ~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kiguchi/miniconda3/envs/nanomotif/lib/python3.12/site-packages/polars/dataframe/frame.py", line 1107, in __getitem__
    return get_df_item_by_key(self, key)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kiguchi/miniconda3/envs/nanomotif/lib/python3.12/site-packages/polars/_utils/getitem.py", line 146, in get_df_item_by_key
    return _select_rows(selection, row_key)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kiguchi/miniconda3/envs/nanomotif/lib/python3.12/site-packages/polars/_utils/getitem.py", line 301, in _select_rows
    return _select_rows_by_index(df, indices)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kiguchi/miniconda3/envs/nanomotif/lib/python3.12/site-packages/polars/_utils/getitem.py", line 317, in _select_rows_by_index
    return df._from_pydf(df._df.gather_with_series(key._s))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
pyo3_runtime.PanicException: implementation error`

**Comment by SorenHeidelbach** [comment]
Thank you for using Nanomotif

The issue seems to be related to polars version (#53). Could you post polars version? 

Best Søren

**Comment by YuyaKiguchi** [comment]
Thanks all.
I could successfully got the results by downgrading of polars version from 0.20.31 to 0.20.4.

**Comment by SebastianDall** [comment]
Fixed in PR #62
